src/Context.Tests/Extensions/MongoDatabaseExtensionsTests.cs

```diff
@@ -0,0 +1,144 @@
+﻿using MongoDB.Bson;
+using MongoDB.Driver;
+using Squadron;
+using Xunit;
+
+namespace MongoDB.Extensions.Context.Tests.Extensions
+{
+    public class MongoDatabaseExtensionsTests : IClassFixture<MongoResource>
+    {
+        private readonly IMongoDatabase _mongoDatabase;
+
+        public MongoDatabaseExtensionsTests(MongoResource mongoResource)
+        {
+            _mongoDatabase = mongoResource.CreateDatabase();
+        }
+
+        #region EnableProfiling Tests
+
+        [Fact]
+        public void EnableProfiling_EnableDefaultProfileLevel_EnabledProfilingOnLevel2()
+        {
+            // Arrange
+
+            // Act
+            _mongoDatabase.EnableProfiling();
+
+            // Assert
+            var profileStatusCommand = new BsonDocument("profile", BsonValue.Create(null));
+            BsonDocument result = _mongoDatabase.RunCommand<BsonDocument>(profileStatusCommand);
+
+            Assert.Equal(
+                "{ \"was\" : 2, \"slowms\" : 100, \"sampleRate\" : 1.0, \"ok\" : 1.0 }",
+                result.ToString());
+        }
+
+        [Fact]
+        public void EnableProfiling_EnableAllProfileLevel_EnabledProfilingOnLevel2()
+        {
+            // Arrange
+
+            // Act
+            _mongoDatabase.EnableProfiling(ProfileLevel.All);
+
+            // Assert
+            var profileStatusCommand = new BsonDocument("profile", BsonValue.Create(null));
+            BsonDocument result = _mongoDatabase.RunCommand<BsonDocument>(profileStatusCommand);
+
+            Assert.Equal(
+                "{ \"was\" : 2, \"slowms\" : 100, \"sampleRate\" : 1.0, \"ok\" : 1.0 }",
+                result.ToString());
+        }
+
+        [Fact]
+        public void EnableProfiling_EnableSlowOperationsOnlyProfileLevel_EnabledProfilingOnLevel1()
+        {
+            // Arrange
+
+            // Act
+            _mongoDatabase.EnableProfiling(ProfileLevel.SlowOperationsOnly);
+
+            // Assert
+            var profileStatusCommand = new BsonDocument("profile", BsonValue.Create(null));
+            BsonDocument result = _mongoDatabase.RunCommand<BsonDocument>(profileStatusCommand);
+
+            Assert.Equal(
+                "{ \"was\" : 1, \"slowms\" : 100, \"sampleRate\" : 1.0, \"ok\" : 1.0 }",
+                result.ToString());
+        }
+
+        [Fact]
+        public void EnableProfiling_DisableProfileLevel_EnabledProfilingOnLevel1()
+        {
+            // Arrange
+
+            // Act
+            _mongoDatabase.EnableProfiling(ProfileLevel.Off);
+
+            // Assert
+            var profileStatusCommand = new BsonDocument("profile", BsonValue.Create(null));
+            BsonDocument result = _mongoDatabase.RunCommand<BsonDocument>(profileStatusCommand);
+
+            Assert.Equal(
+                "{ \"was\" : 0, \"slowms\" : 100, \"sampleRate\" : 1.0, \"ok\" : 1.0 }",
+                result.ToString());
+        }
+
+        #endregion
```

Comment:
I like to put the name of the region after #endregion. makes it easier to co-relate

Reply:
Yep, corrected it.